src/components/Modals/ContactFormModal.jsx

```diff
@@ -1,42 +1,34 @@
-import React, { useState, useEffect, useContext } from 'react'
+import React, { useState, useContext } from 'react'
 import { useNavigate, useParams } from 'react-router-dom'
 
-import { useClient } from 'cozy-client'
+import { useClient, useQuery } from 'cozy-client'
 import { FixedDialog } from 'cozy-ui/transpiled/react/CozyDialogs'
 import Alerter from 'cozy-ui/transpiled/react/Alerter'
 import Button from 'cozy-ui/transpiled/react/Buttons'
 import { useI18n } from 'cozy-ui/transpiled/react/I18n'
-import { buildContactsQueryById } from '../../queries/queries'
 
-import SelectedGroupContext from '../Contexts/SelectedGroup'
-import ContactForm, { getSubmitContactForm } from '../ContactCard/ContactForm'
-import { createContact, updateContact } from '../../connections/allContacts'
-import { addGroupToContact } from '../../helpers/contacts'
-import { hasSelectedGroup } from '../../helpers/groups'
+import { buildContactsQueryById } from '../../queries/queries'
+import SelectedGroupContext from '../../components/Contexts/SelectedGroup'
+import ContactForm, {
+  getSubmitContactForm
+} from '../../components/ContactCard/ContactForm'
+import { createOrUpdateContact } from '../../connections/allContacts'
 
 const ContactFormModal = () => {
   const navigate = useNavigate()
   const { t } = useI18n()
   const client = useClient()
   const { contactId } = useParams()
 
-  useEffect(() => {
-    const fetchContactById = async () => {
-      if (contactId && !contactInForm) {
-        const contactsQueryById = buildContactsQueryById(contactId)
-        const resultContactsQueryById = await client.fetchQueryAndGetFromState({
-          definition: contactsQueryById.definition,
-          options: contactsQueryById.options
-        })
-        setContactInForm(resultContactsQueryById.data)
-      }
-    }
-    fetchContactById()
-  }, [client, contactId, contactInForm])
-
   const { selectedGroup } = useContext(SelectedGroupContext)
   const [isFormBeingSubmitted, setIsFormBeingSubmitted] = useState(false)
-  const [contactInForm, setContactInForm] = useState()
+
+  // Tip while waiting for `getById` to be fixed
+  const contactsQueryById = buildContactsQueryById(contactId || 'undefined')
```

Comment:
I don't understand the 'undefined'. It is to set in the query `enabled` to `false`where there is no contactId ? What is the issue with `getById` ?

Reply:
+1 pour un link dans le code vers l'issue.
j'ai l'impression qu'on fait bien `enabled: Boolean(contactId)` pourtant ?

Reply:
Il y a en effet un soucis ici, le cas problème est la création d'un nouveau contact, `contactId` est donc undefined et `getById` plante.
Ce qui ne devrait pas suite au fix https://github.com/cozy/cozy-client/commit/c5b602256f3e4cd747734fa213500191eeb2e3c9 ...
Je vais creuser

Reply:
>j'ai l'impression qu'on fait bien enabled: Boolean(contactId) pourtant ?

En effet 🤷

Reply:
Attention aussi parce que Boolean('undefined') renvoie true.